### ISSUE TITLE
Display message errors in chat log

### ISSUE THREAD

**Issue by ge0rg** [tigase/siskin-im#1]
When delivery of a message fails, this is not visible to the user. However, it would be much better to have an indication that the message was not delivered.

This indication should NOT be displayed if a Receipt has been received for that message already, and it should be reset if a Receipt arrives after the fact.

This is how it looks in yaxim:
![image](https://user-images.githubusercontent.com/165635/62486388-6d2fda80-b7bf-11e9-8ceb-565fde2df36c.png)

P.S: you can test the behavior by sending messages to xmpp:reject@yax.im

**Comment by hantu85** [comment]
If the message is not delivered then it is marked by SiskinIM in the chat (if was stamped with ID, the response had a body and it was possible to be identified). In any other case message, the message is skipped. 

If the message does not have a body element then in most cases it will not be stored in the offline storage anyway, so with iOS client (being mostly offline), the error would not be delivered anyway.

**Comment by ge0rg** [comment]
I've tested the behavior by sending a message from Siskin to reject@yax.im, which will automatically send an error response to the message, referencing the original message id. However, Siskin didn't show any annotation to the message. Feel free to test yourself with the bot.

**Comment by hantu85** [comment]
As I've mentioned, Siskin adds an error if the response has a body, while reject@yax.im does not contain body element as I've checked.

**Comment by ge0rg** [comment]
I haven't yet seen message errors with a body element in the wild, and I'm pretty sure that would be considered as non-standard by most. Have a look at [RFC6120 §8.3](https://xmpp.org/rfcs/rfc6120.html#stanzas-error) for the rules around message errors, they only suggest an `<error>` element (which can contain a text description of the error), and optionally returning back the original message XML.

**Comment by hantu85** [comment]
I know RFC and I also know that in most cases there is no specific `<body/>` element, but most of the implementations return the original message XML which contains `<body/>` element.

I've already mentioned, the message of type error without body element most likely will be dropped if the client is disconnected as message will not be saved to the offline storage by the server unless we add a hint to store a message. But this as well is not within RFC and would be considered non-standard.

So we can improve the client handling of the error (and we will), however, it will not improve the storage of error message while the client is not connected.

**Comment by ge0rg** [comment]
Yes, I am aware of the sad situation with message errors that do not contain a body. I've actually just [started a thread](https://mail.jabber.org/pipermail/standards/2019-August/036332.html) about this problem, in the hope to fix the respective standards. Feel free to chime in there :)
